### PR TITLE
chore: reimplement filters

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/web/src/uiUtils.ts [hunk 1]
```diff
@@ -76,3 +78,17 @@ export function copy(text: string) {
   document.execCommand('copy');
   textArea.remove();
 }
+
+export function useSetting<S>(name: string, defaultValue: S): [S, React.Dispatch<React.SetStateAction<S>>] {
```
Comment (dgozman) [inline packages/web/src/uiUtils.ts:82]: I didn't find the place that imports this function.

File: packages/trace-viewer/src/ui/watchMode.tsx [hunk 2]
```diff
@@ -148,52 +141,42 @@ export const WatchModeView: React.FC<{}> = ({
       </div>
       <div className='vbox watch-mode-sidebar'>
         <Toolbar>
-          <div className='section-title' style={{ cursor: 'pointer' }} onClick={() => setSettingsVisible(false)}>Tests</div>
-          <ToolbarButton icon='play' title='Run all' onClick={() => runTests(visibleTestIds)} disabled={isRunningTest}></ToolbarButton>
-          <ToolbarButton icon='debug-stop' title='Stop' onClick={() => sendMessageNoReply('stop')} disabled={!isRunningTest}></ToolbarButton>
-          <ToolbarButton icon='refresh' title='Reload' onClick={() => refreshRootSuite(true)} disabled={isRunningTest}></ToolbarButton>
+          <img src='icon-32x32.png' />
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/watchMode.tsx:144]: Is this icon missing from the PR?

**Review by dgozman — APPROVED** [review]

File: packages/web/src/uiUtils.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/web/src/uiUtils.ts:82]: This is for the future use

File: packages/trace-viewer/src/ui/watchMode.tsx (hunk 2 above)
Comment (pavelfeldman) [reply packages/trace-viewer/src/ui/watchMode.tsx:144]: We had it

**Pull request merged**